commit 585e751ad6cbf1d2fe212cca0004534ec9c8b472
Author: Kamil Trzciński <ayufan@ayufan.eu>
Date:   Thu Dec 12 20:29:27 2024 +0100

    Fix mismatched types on arm64 for `src/crypt.rs`
    
      --> /src/proxmox/proxmox-sys/src/crypt.rs:74:67
       |
    74 |         if data.output.first().is_none() || Some(&('*' as i8)) == data.output.first() {
       |                                                                   ^^^^^^^^^^^^^^^^^^^ expected `Option<&i8>`, found `Option<&u8>`
       |
       = note: expected enum `std::option::Option<&i8>`
                  found enum `std::option::Option<&u8>`

diff --git a/proxmox-sys/src/crypt.rs b/proxmox-sys/src/crypt.rs
index 3313f668..a11c19f1 100644
--- a/proxmox-sys/src/crypt.rs
+++ b/proxmox-sys/src/crypt.rs
@@ -71,7 +71,7 @@ pub fn crypt(password: &[u8], salt: &[u8]) -> Result<String, Error> {
         // > output field of their data argument, and crypt writes an invalid hash to its static
         // > storage area.  This string will be shorter than 13 characters, will begin with a ‘*’,
         // > and will not compare equal to setting.
-        if data.output.first().is_none() || Some(&('*' as i8)) == data.output.first() {
+        if data.output.first().is_none() || Some(&('*' as libc::c_char)) == data.output.first() {
             bail!("internal error: crypt_r returned invalid hash");
         }
         CStr::from_ptr(&data.output as *const _)
@@ -133,7 +133,7 @@ pub fn crypt_gensalt(prefix: &str, count: u64, rbytes: &[u8]) -> Result<String,
     // while it states that this is "in addition" to returning a null pointer, this isn't how
     // `crypt_r` seems to behave (sometimes only setting an invalid hash) so add this here too just
     // in case.
-    if output.first().is_none() || Some(&('*' as i8)) == output.first() {
+    if output.first().is_none() || Some(&('*' as libc::c_char)) == output.first() {
         bail!("internal error: crypt_gensalt_rn could not create a valid salt");
     }
 
